core/client/example/go.mod

```diff
@@ -0,0 +1,50 @@
+module mykwilapp
+
+go 1.22.1
+
+require (
+	// this kuneiform rev uses the update sqlparser on kwil-db's main:
+	github.com/kwilteam/kuneiform v0.6.1-0.20240418203939-18f750d293bf
+	github.com/kwilteam/kwil-db/core v0.1.2
```

Comment:
These two requires for a basic app got me thinking about the `parse` module again.  If an app has to import the _main_ module as well as the `core` module to work with .kf files, that could be unfortunate for an app developer.

Reply:
I'm not opposed to that, however what is the difference between a developer importing `core` and `main` vs `core` and `parse`?

Reply:
There are some considerations I hadn't really though about clearly until doing this:

- Independent versioning is more important here since the main module version tags refer to our product version, which is not a package.  A package version following semantic versioning is more important.
- sequestering irrelevant module dependencies is a "good thing".  e.g. writing an app that uses kwil that has postgres and cometbft packages in it's go.mod is confusing and potentially frustrating

These were reasons when we made the  `parse` module initially, but we also had the kuneiform circular require to defeat.

I think it would be better to have `parse` module with `parse/kuneiform` packages.